src/components/HydrantList.js

```diff
@@ -23,10 +26,53 @@ const styles = theme => ({
 
 class HydrantList extends React.Component {
 
-  state = {
-    open: true,
+
+  constructor(props){
+    super(props)
+    this.state = {
+      open: true,
+      prefix: this.getPrefix(props.trailHydrants),
+    }
   }
 
+  componentWillReceiveProps(nextProps) {
+    const prefix = this.getPrefix(nextProps.trailHydrants)
+    this.setState({
+        prefix
+    })
+
+  }
+
+  getPrefix = (hydrants) => {
+    if (hydrants.size > 1) {
+      return hydrants.reduce((accum, val) => {
+        let name = val.get('name')
+        for (let i = 0; i < accum.length; i++) {
+          if (accum[i] !== name[i]) {
+            return accum.slice(0, i)
+          }
+        };
+        return accum
+      }, hydrants.getIn([0,'name']) )
+    }
+    return ''
+  }
+
+  applyPrefix = () => {
+    const { modifyHydrant, trailHydrants } = this.props
+    const { prefix } = this.state
+    const oldPrefix = this.getPrefix(trailHydrants)
+
+
+    trailHydrants.forEach((h) => {
+      const newName = h.get('name').replace(oldPrefix, prefix)
+      modifyHydrant(h.get('id'), { name: newName })
```

Comment:
This would be more efficient with the `dataImported` action (which is due for a renaming too since it's really any bulk update now).